### PR TITLE
feat: Show version info in debug window (#12000)(CP: 2.8)

### PR DESCRIPTION
**Pull request by mshabarov**
Show version info in debug window and allow copying it to clipboard.

(cherry picked from commit 5cb22904929ba33133c12a8dc2f55488af986e56)

Needed for https://github.com/vaadin/flow/pull/14871

Part-of https://github.com/vaadin/flow/issues/14633

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/internal/BrowserLiveReload.java [hunk 1]
```diff
@@ -82,4 +82,9 @@ enum Backend {
      */
     void reload();
 
+    /**
+     * Called when any message is received through the connection.
+     */
+    void onMessage(String msg);
+
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/internal/BrowserLiveReload.java]: Done

File: flow-server/src/main/java/com/vaadin/flow/internal/DebugWindowConnection.java [hunk 2]
```diff
@@ -141,6 +157,11 @@ public void reload() {
     }
 
     private static Logger getLogger() {
-        return LoggerFactory.getLogger(BrowserLiveReloadImpl.class.getName());
+        return LoggerFactory.getLogger(DebugWindowConnection.class.getName());
+    }
+
+    @Override
+    public void onMessage(String message) {
+        // No messages supported yet
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/internal/DebugWindowConnection.java]: Done

File: flow-server/pom.xml [hunk 3]
```diff
@@ -74,6 +74,11 @@
             <artifactId>jsoup</artifactId>
             <version>${jsoup.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-server/pom.xml]: Done

File: flow-server/src/main/java/com/vaadin/flow/internal/DebugWindowConnection.java [hunk 4]
```diff
@@ -110,6 +114,18 @@ public void onConnect(AtmosphereResource resource) {
         atmosphereResources.add(new WeakReference<>(resource));
         resource.getBroadcaster().broadcast("{\"command\": \"hello\"}",
                 resource);
+
+        send(resource, "serverInfo", new ServerInfo());
+    }
+
+    private void send(AtmosphereResource resource, String command,
+            Object data) {
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/internal/DebugWindowConnection.java]: Done

File: flow-client/src/main/resources/META-INF/frontend/VaadinDevmodeGizmo.ts [hunk 5]
```diff
@@ -9,6 +18,8 @@ export class VaadinDevmodeGizmo extends LitElement {
   static GREY_HSL = css`0, 0%, 50%`;
   static YELLOW_HSL = css`38, 98%, 64%`;
   static RED_HSL = css`355, 100%, 68%`;
+  static MAX_LOG_ROWS = 1000;
+  static copyO = svg`<svg style="width: 16px; height: 16px"><g id="copy-o"><path d="M13 3h-3l-3-3h-7v13h6v3h10v-10l-3-3zM7 1l2 2h-2v-2zM1 12v-11h5v3h3v8h-8zM15 15h-8v-2h3v-9h2v3h3v8zM13 6v-2l2 2h-2z"></path></g></svg>`;
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-client/src/main/resources/META-INF/frontend/VaadinDevmodeGizmo.ts]: Done

**Review by caalador — APPROVED** [review]

**Comment by caalador** [comment]
```
java.lang.AssertionError: Serializable not implemented by the following classes and interfaces: ,
com.vaadin.flow.internal.DebugWindowData
```

**Comment by mshabarov** [comment]
> java.lang.AssertionError: Serializable not implemented by the following classes and interfaces: ,
com.vaadin.flow.internal.DebugWindowData

Ah, I thought that interfaces do not need it

**Pull request merged**